serverless.js

```diff
@@ -71,11 +72,32 @@ class Template extends Component {
     return outputs
   }
 
-  async remove() {
+  async remove(inputs = {}) {
     this.context.status('Removing')
 
     this.context.debug('Flushing template state and removing all components.')
-    await syncState({}, this)
```

Comment:
Why `syncState({}, this)` is not enough to properly remove components?

Reply:
Because  syncState() just remove all components at the same time, not analysis the dependency tree. For some relative components, should remove the relative component firstly, otherwise it will fail to remove.

Reply:
> Because syncState() just remove all components at the same time,

But you've fixed that in that PR here: https://github.com/serverless/template/pull/17/files/bc2003dd824351b04cc40cb3ff9f3c1e436985c5#diff-14ef994fa167d355bdaf9b44add54b87R337

So my question is why is that fix not enough (?)